lightwood/data/splitter.py

```diff
@@ -19,16 +22,27 @@ def splitter(data: pd.DataFrame, k: int, tss: TimeseriesSettings) -> List[pd.Dat
             folds = np.array_split(data, k)
         else:
             gcols = tss.group_by
-            all_group_combinations = list(product(*[data[gcol].unique() for gcol in tss.group_by]))
-            folds = [pd.DataFrame() for _ in range(k)]
-            for group in all_group_combinations:
-                subframe = data
-                for idx, gcol in enumerate(gcols):
-                    subframe = subframe[subframe[gcol] == group[idx]]
+            folds = grouped_ts_splitter(data, k, gcols)
 
-                subfolds = np.array_split(subframe, k)
+    return folds
+
+
+def grouped_ts_splitter(data: pd.DataFrame, k: int, gcols: List[str]):
+    """
+    Splitter for grouped time series tasks, where there is a set of `gcols` columns by which data is grouped.
+    Each group yields a different time series, and the splitter generates `k` folds from `data`,
+    with equally-sized sub-series for each group.
+    """
+    all_group_combinations = list(product(*[data[gcol].unique() for gcol in gcols]))
```

Comment:
I'm actually not fully aware what product(*<var>) does. Can you elaborate?

Reply:
It yields the cartesian product of all observed values for each `group_by` column.

So, if `gby_col_a` has `A, B` as possible values, and `gby_col_b` has `C, D` as possible values, this line of code determines that all possible time series are given by the following combinations:

`[(A, C), (B, C), (A, D), (B, D)]`

Reply:
Great - be nice to make a small note of that in the docstring explicitly with the phrase you used earlier ("It yields the cartesian product of all observed values for each group_by column.")